src/Runner.Common/JobServer.cs

```diff
@@ -125,6 +157,46 @@ private void CheckConnection()
             }
         }
 
+        private void InitializeWebsocketClient(TimeSpan delay)
+        {
+             if (_serviceEndpoint.Authorization != null &&
+                 _serviceEndpoint.Authorization.Parameters.TryGetValue(EndpointAuthorizationParameters.AccessToken, out var accessToken) &&
+                 !string.IsNullOrEmpty(accessToken))
+            {
+                if (_serviceEndpoint.Data.TryGetValue("FeedStreamUrl", out var feedStreamUrl) && !string.IsNullOrEmpty(feedStreamUrl))
+                {
+                    // let's ensure we use the right scheme
+                    feedStreamUrl = feedStreamUrl.Replace("https://", "wss://").Replace("http://", "ws://");
+                    Trace.Info($"Creating websocket client ..." + feedStreamUrl);
+                    this._websocketClient = new ClientWebSocket();
+                    this._websocketClient.Options.SetRequestHeader("Authorization", $"Bearer {accessToken}");
+                    this._websocketConnectTask = Task.Run(async () =>
+                    {
+                        try
+                        {
+                            Trace.Info($"Attempting to start websocket client with delay {delay}.");
+                            await Task.Delay(delay);
+                            await this._websocketClient.ConnectAsync(new Uri(feedStreamUrl), default(CancellationToken));
+                            Trace.Info($"Successfully started websocket client.");
+                        }
+                        catch(Exception ex)
+                        {
+                            Trace.Info("Exception caught during websocket client connect, fallback of HTTP would be used now instead of websocket.");
```

Comment:
do we need to set `_websocketClient = null` if we fail to connect?

Reply:
we use that `null` to determine whether to attempt reconnect again, so not here